test/integration/error_log_test.go

```diff
@@ -0,0 +1,124 @@
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+)
+
+func TestLogfiles(t *testing.T) {
+	profile := UniqueProfileName("logfiles")
+	ctx, cancel := context.WithTimeout(context.Background(), Minutes(5))
+	defer Cleanup(t, profile, cancel)
+
+	tests := []struct {
+		command          string
+		args             []string
+		runCount         int // number of times to run command
+		expectedLogFiles int // number of logfiles expected after running command runCount times
+	}{
+		{
+			command:          "start",
+			args:             append([]string{"--memory=2200", "--wait=true"}, StartArgs()...),
+			runCount:         20, // calling this 20 times should create 2 files with 1 greater than 1M
+			expectedLogFiles: 2,
+		}, {
+			command:          "status",
+			runCount:         100,
+			expectedLogFiles: 1,
+		}, {
+			command:          "pause",
+			runCount:         5,
+			expectedLogFiles: 1,
+		}, {
+			command:          "unpause",
+			runCount:         1,
+			expectedLogFiles: 1,
+		}, {
+			command:          "stop",
+			runCount:         1,
+			expectedLogFiles: 1,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.command, func(t *testing.T) {
+			args := []string{test.command, "-p", profile}
+			args = append(args, test.args...)
+			// run command runCount times
+			for i := 0; i < test.runCount; i++ {
+				rr, err := Run(t, exec.CommandContext(ctx, Target(), args...))
+				if err != nil {
+					t.Errorf("failed to clean up: args %q: %v", rr.Command(), err)
+				}
+			}
+			// get log files generated above
+			logFiles, err := getLogFiles(test.command)
+			if err != nil {
+				t.Errorf("failed to find tmp log files: command %s : %v", test.command, err)
+			}
+			// cleanup generated logfiles
+			defer cleanupLogFiles(t, logFiles)
+			// if not the expected number of files, throw err
+			if len(logFiles) != test.expectedLogFiles {
+				t.Errorf("failed to find expected number of log files: cmd %s: expected: %d got %d", test.command, test.expectedLogFiles, len(logFiles))
+			}
+			// if more than 1 logfile is expected, only one file should be less than 1M
+			if test.expectedLogFiles > 1 {
+				foundSmall := false
+				maxSize := 1024 * 1024 // 1M
+				for _, logFile := range logFiles {
+					if int(logFile.Size()) < maxSize && !foundSmall {
+						foundSmall = true
+					} else if int(logFile.Size()) < maxSize && foundSmall {
+						t.Errorf("expected to find only one file less than 1M: cmd %s:", test.command)
+					}
+				}
+			}
+		})
+
+	}
+}
+
+// getLogFiles returns logfiles corresponding to cmd
+func getLogFiles(cmdName string) ([]os.FileInfo, error) {
+	var logFiles []os.FileInfo
+	err := filepath.Walk(os.TempDir(), func(path string, info os.FileInfo, err error) error {
+		if strings.Contains(info.Name(), fmt.Sprintf("minikube_%s", cmdName)) {
+			logFiles = append(logFiles, info)
+		}
+		return nil
+	})
+	return logFiles, err
```

Comment:
I don't think it really makes sense to return `err` in this func, it's not possible for `err` to be anything than `nil`, so we'll just doing an unnecessary error check.

Reply:
I just realized this is in the test file, this isn't a huge deal.

Reply:
Couldn't there be a case where `filepath.Walk` returns an error in which case the `err` would not be nil? Sorry very much a newbie to golang so apologies if this is a stupid question.

Reply:
You are correct, I assumed that the only way to return an error would be from the func you defined as the second argument, but upon inspection of the source code of `filepath.Walk` there is a path where a non-nil error could be returned even if your func always returns `nil`.